src/app/archive-settings/manage-metadata/manage-custom-metadata/manage-custom-metadata.component.ts

```diff
@@ -0,0 +1,136 @@
+/* @format */
+import {
+  Component,
+  ElementRef,
+  OnDestroy,
+  OnInit,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+import { TagsService } from '@core/services/tags/tags.service';
+import { ApiService } from '@shared/services/api/api.service';
+import { TagVO, TagVOData } from '@models/tag-vo';
+import { Subject } from 'rxjs';
+
+@Component({
+  selector: 'pr-manage-custom-metadata',
+  templateUrl: './manage-custom-metadata.component.html',
+  styleUrls: ['./manage-custom-metadata.component.scss'],
+})
+export class ManageCustomMetadataComponent implements OnInit, OnDestroy {
+  @ViewChild('element') public element: ElementRef;
+  public categories: string[] = [];
+  public activeCategory: string | null = null;
+  public values: TagVO[] = [];
+
+  public dismissEvent = new Subject<number>();
+
+  public tagsList: TagVO[] = [];
+  protected tagMap: Map<string, TagVO[]> = new Map<string, TagVO[]>();
+
+  protected deletedTagIds: number[] = [];
+
+  protected windowListener: () => void;
+
+  constructor(private renderer: Renderer2, private tag: TagsService) {
+    this.windowListener = this.renderer.listen(
+      'window',
+      'click',
+      (e: Event) => {
+        if (e.target && !this.element.nativeElement.contains(e.target)) {
+          this.dismissEvent.next(-1);
+        }
+      }
+    );
+  }
+
+  ngOnInit(): void {
+    this.getTagsFromTagService();
+    this.tagMap.clear();
+    this.reloadTagMap();
+  }
+
+  ngOnDestroy(): void {
+    if (this.windowListener) {
+      this.windowListener();
+    }
+  }
+
+  public setActiveCategory(category: string): void {
+    if (!this.tagMap.has(category)) {
+      this.activeCategory = null;
+      this.values = [];
+      return;
+    }
+    this.activeCategory = category;
+    this.values = this.tagMap.get(category);
+  }
+
+  public async refreshTagsInPlace() {
+    await this.tag.resetTags();
+    this.getTagsFromTagService();
+    this.tagMap.clear();
+    this.reloadTagMap();
+    this.setActiveCategory(this.activeCategory);
+  }
+
+  public identifyTag(index: number, item: TagVO): number {
```

Comment:
I don't know if any conventions exist for this in Angular or JS in general, but yes the `index` argument is only filled in because the value we really want is the second argument.